### ISSUE TITLE
"X has changed its name to X"

### ISSUE THREAD

**Issue by NicoWeio** [instaloader/instaloader#1530]
**Describe the bug**
For me, this appears together with #1523, but it might be a more general issue:
Instaloader reports that a user has changed their name, while in the output, the old and new names are identical.

**To Reproduce**
- Extract a session from Firefox
- `instaloader --login=[…] --stories --fast-update […]`

**Expected behavior**
A clear and concise description of what you expected to happen (even if it seems obvious).

**Error messages and tracebacks**
One relevant line: `JSON Query to […]/feed/: Could not find "window._sharedData" in html response.`

<details>
<summary>Full output</summary>

```
Loaded session from /home/nicolai/.config/instaloader/session-[…].
Logged in as […].
JSON Query to […]/feed/: Could not find "window._sharedData" in html response. [retrying; skip with ^C]
JSON Query to […]/feed/: Could not find "window._sharedData" in html response. [retrying; skip with ^C]
Trying to find profile […] using its unique ID […].
Profile […] has changed its name to […].
JSON Query to […]/feed/: Could not find "window._sharedData" in html response. [retrying; skip with ^C]
JSON Query to […]/feed/: Could not find "window._sharedData" in html response. [retrying; skip with ^C]
Profile […] does not exist.
The most similar profiles are: […].
Trying again anonymously, helps in case you are just blocked.
Warning: […] will be downloaded anonymously ("Profile […] does not exist.
The most similar profiles are: […].").
Downloading anonymously: […]
[1/1] Downloading profile […]
[…]/2020-05-02_10-12-06_UTC_profile_pic.jpg already exists
Retrieving posts from profile […].
[ 1/40] […]/2022-05-13_05-55-51_UTC.jpg [[…]] json 
[ 2/40] […]/2022-01-09_18-04-40_UTC.jpg exists [[…]] unchanged json 
Saved session to /home/nicolai/.config/instaloader/session-[…].

Errors or warnings occurred:
Profile […] has changed its name to […].
Warning: […] will be downloaded anonymously ("Profile […] does not exist.
The most similar profiles are: […].").
```

</details>

**Instaloader version**
4.9

**Comment by Showmaster3000** [comment]
Is it possible that the profile you download blocked you?

**Comment by NicoWeio** [comment]
I don't think so. The profile loads fine in the Instagram app, with the username unchanged.

**Comment by Showmaster3000** [comment]
The problem still exists with current master. Since about 3 hours, one of my accounts has the same issue. Downloading the profiles with another account works fine.
I recreated the session of the failing account, but no luck. The profiles are downloaded anonymously, but this doesn't work for stories.

**Comment by Greyb0w1er** [comment]
> Another thing I noticed is that profiles load fine when providing the password upon execution, i.e. without a session file. However, the newly created session files don't work on the next invocation either.

Same here, loaded previuos session and it got stuck on error `JSON Query to […]/feed/: Could not find "window._sharedData" in html response.`, tried login on execution and it worked like a charm, every time

**Comment by SungMatt** [comment]
Same issue with me too -- happened after I updated to 4.9.

**Comment by pirinoa** [comment]
Yeah, same with no config or version changes for me. macOS instaloader on 4.9, tried 2 different session logins, same error each time. Default quality works, the second you add `--login=` to the string the errors appear. Tried 3 different target accounts, same issue that started happening a few weeks ago!

Used this command for months then the errors started coming up `Could not find "window._sharedData" in html response. [retrying; skip with ^C]` - no issues with IOS app, macOS web logins and users for these accounts.

**Comment by Showmaster3000** [comment]
@pirinoa Did you check #1553? There is a patch by another user which solves the problem with `window._sharedData`.

**Comment by pirinoa** [comment]
Thanks so much @Showmaster3000 - applied the patch and the window error's gone! 👏

**Comment by NicoWeio** [comment]
I fixed the nonsensical error message(s) in #1536,
and the underlying issue has been fixed in #1561.

In short, `pip install --upgrade instaloader` should resolve these issues.
